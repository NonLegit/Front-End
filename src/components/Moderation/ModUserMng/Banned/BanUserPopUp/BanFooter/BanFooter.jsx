import { Typography, Box } from '@mui/material';
import { StyledBox } from './styles';
import { RedditButton } from '../../../../styles';

/**
 * Ban Footer
 * @component
 * @property  {function} handleClickCloseBan close the popup when a button is clicked
 * @property  {function} handleBan operate the function of banning user and send data to backend when a button is clicked
 * @return {React.Component} - Ban Footer component
 */

function BanFooter(props) {
  const { handleClickCloseBan, handleBan } = props;

  return (
    <StyledBox>
      <Typography>
        â€¢ Visible to banned user
      </Typography>
      <Box>
        <RedditButton
          fontSize="15px"
          padding="0px 24px 0px 24px"
          fontWeight="bold"
          variant="outlined"
          onClick={handleClickCloseBan}
          disableRipple
          disableFocusRipple
        >
          Cancel
        </RedditButton>
        <RedditButton
          sx={{ marginLeft: 1 }}
          fontSize="14px"
          padding="0px 24px 0px 24px"
          fontWeight="bold"
          variant="contained"
          disableRipple
          disableFocusRipple
          onClick={handleBan}
        >
          Ban user
        </RedditButton>
      </Box>
    </StyledBox>
  );
}

export default BanFooter;
