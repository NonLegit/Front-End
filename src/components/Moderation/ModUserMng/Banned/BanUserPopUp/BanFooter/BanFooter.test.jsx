import renderer from 'react-test-renderer';
import BanFooter from './BanFooter';

test('test snapshot', async () => {
  const tree = renderer.create(<BanFooter />).toJSON();
  expect(tree).toMatchSnapshot();
});
