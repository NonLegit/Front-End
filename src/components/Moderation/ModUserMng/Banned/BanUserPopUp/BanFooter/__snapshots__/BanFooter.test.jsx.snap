// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<div
  className="MuiBox-root css-1ucnzwz"
>
  <p
    className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  >
    â€¢ Visible to banned user
  </p>
  <div
    className="MuiBox-root css-0"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1bny8zm-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      fontSize="15px"
      fontWeight="bold"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      padding="0px 24px 0px 24px"
      tabIndex={0}
      type="button"
    >
      Cancel
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1dubyk3-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      fontSize="14px"
      fontWeight="bold"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      padding="0px 24px 0px 24px"
      tabIndex={0}
      type="button"
    >
      Ban user
    </button>
  </div>
</div>
`;
